docs/plugin_developers_guide/02_persisting_data.rst

```diff
@@ -0,0 +1,98 @@
+
+The following is an example taken from a filter plugin
+
+.. code-block:: C
+
+    /**
+     * Pass saved plugin configuration
+     */
+    void plugin_start(PLUGIN_HANDLE *handle, string& data)
+    {
+    FILTER_INFO *info = (FILTER_INFO *) handle;
+
+        Sigma *filter = info->handle;
+        filter->loadState(data);
+    }
+
+
+The plugin_start entry point is called after the plugin_init entry point has been called and before any data is sent to or requested from the plugin.
+
+If there is no data persisted for the plugin, because it has never be previously run or not persisted any data via shutdown, an empty string will be passed to the plugin.
+
+Persisting Data
+###############
+
+As discussed above, data is persisted upon shutdown of the plugin, this is achieved by altering the signature of the plugin_shutdown entry point to return a string. The following example is taken from a filter plugin, but persistence of state can be done for any filter type.
+
+.. code-block:: C
+
+   /**
+    * Call the shutdown method in the plugin
+    */
+   std::string plugin_shutdown(PLUGIN_HANDLE *handle)
+   {
+      FILTER_INFO *info = (FILTER_INFO *) handle;
+      Sigma *filter = info->handle;
+
+      // Fetch the state information to persist
+      std::string state = filter->saveState();
+
+      // Delete the filter
+      delete filter;
+      return state;
+   }
+
+.. note::
+
+   Data is persisted only on orderly shutdown, therefore it is important to avoid uncontrolled shutdowns if you are using this persistence scheme. It is always best to do an orderly shutdown of Fledge, but even more so in these circumstances.
+
+Viewing Persisted Data
+######################
+
+The Fledge user interface has a mechanism that allows features designed for a pipeline developer to be enabled via the settings menu. One of these features allows persisted data to be viewed. Once developer features are enabled a new tab will appear in the configuration of the service. In this *Developer* tab the persisted data can be viewed for the plugins in the pipeline.
+
++-------------+
+| |persist_1| |
++-------------+
+
+When multiple plugin in the pipeline support persistence, the particular plugin can be selected from the drop down menu in the left of the tab.
+
+A menu exists that may be accessed by clicking on the the dots to the right of the *Developer* tab.
+
++-------------+
+| |persist_2| |
++-------------+
+
+This menu allows the import and export of the persisted data to a file. This is useful when debugging pipelines to alter the state of the plugin. The import function is only available if the plugin is shutdown.
+
+You can also delete the persisted data, allowing you to purge the saved state of the plugin and forcing it back to initial operating conditions.
```

Comment:
We can also point to its REST API endpoints, which are available at the following URL
https://fledge-iot.readthedocs.io/en/develop/rest_api_guide/05_RESTdeveloper.html#view-plugin-persisted-data